recidiviz/aggregate_ingest/fl/fl_aggregate_ingest.py

```
@@ -0,0 +1,72 @@
+# Recidiviz - a platform for tracking granular recidivism metrics in real time
+# Copyright (C) 2018 Recidiviz, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+# =============================================================================
+"""Parse the FL Aggregated Statistics PDF."""
+import locale
+from typing import Dict
+
+import pandas as pd
+import tabula
+
+from recidiviz.aggregate_ingest import aggregate_ingest_utils
+
+
+def parse(filename: str) -> Dict[str, pd.DataFrame]:
+    _setup()
+
+    return {
+        'fl_county_adp': _parse_table_1(filename)
+    }
+
+
+def _parse_table_1(filename: str) -> pd.DataFrame:
+    """Parses the FL County ADP - Table 1 in the PDF."""
+    part1 = tabula.read_pdf(
+        filename,
+        pages=[3],
```

Is there a more dynamic way to select where to start and end in Tabula?

---

Yeah there's less structure to latch onto in a raw PDF. I believe these are really consistent, changing very slowly. But I'm worried about tables that may be of variable length and wind up spilling into a new page on occasion. This could happen if the row scheme is something that changes, like a table oriented around the kinds of releases or charges instead of county. It could also happen if there's a rendering mistake and a table goes too far horizontally and breaks to a newline. 

I suppose we can wait to see how this performs on historical reports to generate bugs if they exist.

---

Looking through the historical PDFs for 7 states, I don't see any cases of malformed tables. As we're writing this parsing logic, I think we should design for cases that fail loudly when a PDF can't be correctly parsed. In this case, future changes will parse the pages 4+ which would fail if this table leaked onto that page.

I think it makes sense to move forward with hard coding the pages here. I've made #684 to track adding guards to ensure that each PDF matches the expected requirements.